app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/ApplicationController.scala

```diff
@@ -62,24 +64,44 @@ class ApplicationController @Inject() (
       }
   }
 
+  private def isEnrolmentInUse(service: Service, loggedInUser: LoggedInUserWithEnrolments)(implicit
+    hc: HeaderCarrier
+  ): Future[Option[ExistingEori]] = {
+    val groupId = loggedInUser.groupId.getOrElse(throw MissingGroupId())
+
+    groupEnrolment.groupIdEnrolments(groupId).flatMap { groupEnrolments =>
+      existingEoriForUserOrGroup(loggedInUser, groupEnrolments) match {
+        case Some(existingEori) => enrolmentStoreProxyService.isEnrolmentInUse(service, existingEori)
+        case _                  => Future.successful(None)
+      }
+    }
+  }
+
   private def isUserEnrolledFor(loggedInUser: LoggedInUserWithEnrolments, service: Service): Boolean =
     enrolledForService(loggedInUser, service).isDefined
 
   private def cdsEnrolmentCheck(loggedInUser: LoggedInUserWithEnrolments, groupId: String, service: Service)(implicit
     hc: HeaderCarrier,
     request: Request[_]
   ): Future[Result] =
-    if (isUserEnrolledFor(loggedInUser, Service.cds))
-      Future.successful(Redirect(routes.HasExistingEoriController.displayPage(service)))
-    else
-      groupEnrolment.groupIdEnrolmentTo(groupId, Service.cds).flatMap {
-        case Some(groupEnrolment) if groupEnrolment.eori.isDefined =>
-          cache.saveGroupEnrolment(groupEnrolment).map { _ =>
-            Redirect(routes.HasExistingEoriController.displayPage(service)) // AutoEnrolment
+    isEnrolmentInUse(service, loggedInUser).flatMap {
+      eoriFromUsedEnrolmentOpt =>
+        if (isUserEnrolledFor(loggedInUser, Service.cds) && eoriFromUsedEnrolmentOpt.isEmpty)
+          Future.successful(Redirect(routes.HasExistingEoriController.displayPage(service)))
```

Comment:
can we redirect to unableToUseIdPage when eoriFromUsedEnrolmentOpt is defined

Reply:
Done